lib/api/controllers/adminController.js

```diff
@@ -119,33 +120,33 @@ module.exports = function AdminController (kuzzle) {
    */
   this.deleteIndexes = function (requestObject, context) {
     var modifiedRequestObject = null;
-
     return kuzzle.services.list.readEngine.listIndexes()
       .then(response => {
         var
-          availableIndexes = response.indexes,
-          userRoles = context.token.user.profile.roles,
+          deferred = q.defer(),
           allowedIndexes = [];
 
-        if (availableIndexes.length > 0) {
-          userRoles.forEach(role => {
-            if (role.indexes !== undefined) {
-              _.forEach(role.indexes, (value, roleIndex) => {
-                if (roleIndex.charAt(0) !== '_' && value._canDelete) {
-                  availableIndexes.forEach(availableIndex => {
-                    if ((availableIndex === roleIndex ||
-                      roleIndex === '*' && availableIndex !== internalIndex) &&
-                      _.indexOf(allowedIndexes, availableIndex) === -1
-                    ) {
-                      allowedIndexes.push(availableIndex);
-                    }
-                  });
-                }
-              });
-            }
-          });
-        }
-
+        response.indexes.filter(index => _.includes(requestObject.data.body.indexes, index));
```

Comment:
`Array.filter()` does not mutate the original array, meaning this line does not appear to do anything.